From 8fbbae50d6d2fdae6ba9fea0c1ace9012781c34b Mon Sep 17 00:00:00 2001
From: "W. Michael Petullo" <mike@flyn.org>
Date: Wed, 8 Jan 2020 20:46:46 -0500
Subject: [PATCH] libgd: provide a -full variant

The php7-mod-gd package requires that freetype support exist in libgd,
but this is not included in the default libgd build. In order to allow
a working php7-mod-gd package while keeping the default libgd package
as small as possible, this commit introduces a -full variant of the
libgd package.

Ref: https://github.com/openwrt/packages/issues/10944
Signed-off-by: W. Michael Petullo <mike@flyn.org>

Modified for OpenWrt 19.07 which doesn't use cmake for this package

Signed-off-by: Michal Vasilek <michal.vasilek@nic.cz>
---
 libs/libgd/Makefile | 36 ++++++++++++++++++++++++++++++++++--
 1 file changed, 34 insertions(+), 2 deletions(-)

diff --git a/libs/libgd/Makefile b/libs/libgd/Makefile
index 0b2c998..da32fe8 100644
--- a/libs/libgd/Makefile
+++ b/libs/libgd/Makefile
@@ -26,20 +26,41 @@ PKG_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/package.mk
 
-define Package/libgd
+define Package/libgd/default
   SECTION:=libs
   CATEGORY:=Libraries
   DEPENDS:=+libjpeg +libpng +LIBGD_TIFF:libtiff +LIBGD_FREETYPE:libfreetype
   TITLE:=The GD graphics library
   URL:=https://libgd.github.io/
+endef
+
+define Package/libgd
+  $(call Package/libgd/default)
   MENU:=1
+  DEPENDS+=+LIBGD_TIFF:libtiff +LIBGD_FREETYPE:libfreetype
+  VARIANT:=default
 endef
 
-define Package/libgd/description
+define Package/libgd-full
+  $(call Package/libgd/default)
+  DEPENDS+=+libtiff +libfreetype
+  TITLE+=(full)
+  VARIANT:=full
+endef
+
+define Package/libgd/description/default
   GD is an open source code library for the dynamic creation of images by
   programmers. GD creates PNG, JPEG and GIF images, among other formats.
 endef
 
+Package/libgd/description=$(Package/libgd/description/default)
+
+define Package/libgd-full/description
+  $(call Package/libgd/description/default)
+  .
+  This variant of the libgd package is compiled will all features enabled.
+endef
+
 define Package/libgd/config
 	if PACKAGE_libgd
 		config LIBGD_TIFF
@@ -69,6 +90,12 @@ CONFIGURE_ARGS += \
 	--with-png=$(STAGING_DIR)/usr \
 	--without-xpm
 
+ifeq ($(BUILD_VARIANT),full)
+	CONFIGURE_ARGS+= \
+		--with-freetype=$(STAGING_DIR)/usr \
+		--with-tiff=$(STAGING_DIR)/usr
+else
+
 ifdef CONFIG_LIBGD_TIFF
 	CONFIGURE_ARGS+= \
 		--with-tiff=$(STAGING_DIR)/usr
@@ -85,6 +112,8 @@ else
 		--without-freetype
 endif
 
+endif
+
 CONFIGURE_VARS += \
 	ac_cv_header_iconv_h=no \
 	am_cv_func_iconv_works=no \
@@ -113,4 +142,7 @@ define Package/libgd/install
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libgd.so.* $(1)/usr/lib/
 endef
 
+Package/libgd-full/install=$(Package/libgd/install)
+
 $(eval $(call BuildPackage,libgd))
+$(eval $(call BuildPackage,libgd-full))
-- 
2.37.1

